crates/miden-lib/src/transaction/script_builder.rs

```diff
@@ -0,0 +1,177 @@
+    ) -> Result<TransactionScript, TransactionScriptBuilderError> {
+        let script = format!(
+            "{} begin {} {} {} end",
+            self.script_includes()?,
+            self.script_expiration(),
+            sections.join(" "),
+            self.script_authentication()
+        );
+
+        let assembler = TransactionKernel::assembler().with_debug_mode(self.in_debug_mode);
+        let tx_script = TransactionScript::compile(script, vec![], assembler)
+            .map_err(TransactionScriptBuilderError::InvalidTransactionScript)?;
+
+        Ok(tx_script)
+    }
+
+    /// Returns a string with the needed include instructions for the script.
+    fn script_includes(&self) -> Result<String, TransactionScriptBuilderError> {
+        let mut includes = String::new();
+
+        let include_script = if self
+            .account_interface
+            .components()
+            .contains(&AccountComponentInterface::BasicFungibleFaucet)
+        {
+            AccountComponentInterface::BasicFungibleFaucet.script_includes()?
+        } else if self
+            .account_interface
+            .components()
+            .contains(&AccountComponentInterface::BasicWallet)
+        {
+            AccountComponentInterface::BasicWallet.script_includes()?
+        } else {
+            return Err(TransactionScriptBuilderError::UnsupportedAccount);
+        };
+
+        includes.push_str(include_script);
+
+        self.account_interface.auth().iter().for_each(|auth_scheme| match auth_scheme {
+            &AuthScheme::RpoFalcon512 { pub_key: _ } => {
+                includes.push_str("use.miden::contracts::auth::basic->auth_tx\n");
+            },
+        });
+
+        if self.expiration_delta.is_some() {
+            includes.push_str("use.miden::tx\n");
+        }
+
+        Ok(includes)
+    }
+
+    /// Returns a string with the authentication procedure call for the script.
+    fn script_authentication(&self) -> String {
+        let mut auth_script = String::new();
+        self.account_interface.auth().iter().for_each(|auth_scheme| match auth_scheme {
+            &AuthScheme::RpoFalcon512 { pub_key: _ } => {
+                auth_script.push_str("call.auth_tx::auth_tx_rpo_falcon512\n");
+            },
+        });
+
+        auth_script
+    }
+
+    /// Returns a string with the expiration delta update procedure call for the script.
+    fn script_expiration(&self) -> String {
+        if let Some(expiration_delta) = self.expiration_delta {
+            format!("push.{expiration_delta} exec.tx::update_expiration_block_delta\n")
+        } else {
+            String::new()
+        }
+    }
+}
+
+// TRANSACTION SCRIPT BUILDER ERROR
+// ============================================================================================
+
+/// Errors related to building a transaction script.
+#[derive(Debug, Error)]
+pub enum TransactionScriptBuilderError {
+    #[error("invalid asset: {0}")]
+    InvalidAsset(AccountIdPrefix),
```

Comment:
This variant's name is a bit generic. I think it could say something like "note asset is not issued by this faucet".

Reply:
I think we should update the name of the error as well. Maybe `IssuanceFaucetMismatch`?